libraries/rush-lib/src/logic/ProjectImpactGraphGenerator.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import fs from 'fs';
+import yaml from 'js-yaml';
+import path from 'path';
+import colors from 'colors/safe';
+import type { RushConfiguration } from '../api/RushConfiguration';
+import type { RushConfigurationProject } from '../api/RushConfigurationProject';
+import { Stopwatch } from '../utilities/Stopwatch';
+
+/**
+ * Project property configuration
+ */
+export interface IProjectConfiguration {
+  includedGlobs: string[];
+  excludedGlobs?: string[];
+  dependentProjects: string[];
+}
+
+/**
+ * The schema of `project-impact-graph.yaml`
+ */
+export interface IFileSchema {
+  globalExcludedGlobs: string[];
+  projects: {
+    [key: string]: IProjectConfiguration;
+  };
+}
+
+export class ProjectImpactGraphGenerator {
+  /**
+   * Full path of repository root
+   */
+  private _repositoryRoot: string;
+
+  /**
+   * Projects within the Rush configuration
+   */
+  private _projects: RushConfigurationProject[];
+
+  /**
+   * Default global excluded globs
+   * Only used if the `<repository_root>/.mergequeueignore` does not exist
+   */
+  private _DefaultGlobalExcludedGlobs: string[] = ['common/autoinstallers/**'];
+
+  /**
+   * Get repositoryRoot and load projects within the rush.json
+   */
+  public constructor(rushConfiguration: RushConfiguration) {
+    this._repositoryRoot = rushConfiguration.rushJsonFolder;
+    this._projects = rushConfiguration.projects;
+  }
+
+  /**
+   * Load global excluded globs
+   * @param repositoryRoot
+   */
+  private _loadGlobalExcludedGlobs(repositoryRoot: string): string[] | undefined {
+    const filePath: string = path.join(repositoryRoot, '.mergequeueignore');
```

Comment:
```suggestion
    const filePath: string = `${repositoryRoot}/.mergequeueignore`;
```

Reply:
I'm not convinced anyone actually cares if paths have mixed slashes.

`path.join` is absolutely more expensive than a simple string concatenation, and it doesn't offer any non-aesthetic benefits if we know the path doesn't contain any `..` or `.` folders.